contracts/token/airdrop/AirdropSeason1.sol

```diff
@@ -0,0 +1,111 @@
+    event ClaimAndStaked(
+        address indexed user,
+        uint128 totalGrant,
+        address indexed delegate,
+        IAirdropSingleSidedStaking.Lock lock
+    );
+    event Claimed(address indexed user, uint128 totalGrant);
+
+    // ========================================== CONSTRUCTOR ==========================================
+    /**
+     * @notice Initiate the contract with a merkle tree root, a token for distribution,
+     *         an expiration time for claims, and the voting vault that tokens will be
+     *         airdropped into.
+     *
+     * @param _token                The token to airdrop
+     * @param _merkleRoot           The merkle root with deposits encoded into it as hash [address, amount]
+     * @param _expiration           The expiration of the airdrop
+     * @param _votingVault          The voting vault to deposit tokens to
+     */
+    constructor(
+        IERC20 _token,
+        bytes32 _merkleRoot,
+        uint256 _expiration,
+        IAirdropSingleSidedStaking _votingVault
+    ) ArcadeAirdropBase(_token, _merkleRoot, _expiration) {
+        if (address(_votingVault) == address(0)) revert AA_ZeroAddress("votingVault");
+
+        votingVault = _votingVault;
+    }
+
+    // ===================================== CLAIM FUNCTIONALITY =======================================
+
+    /**
+     * @notice Claims an amount of tokens in the tree and stakes the tokens in the ArcadeSingleSidedStaking
+     *         contract. The caller can choose who to delegate their voting power to by passing in the
+     *         address of their delegate. If a user already has an active stake in the ArcadeSingleSidedStaking
+     *         voting vault they must use the same delegate address they are already delegating to.
+     *
+     * @param delegate               The address the user will delegate to
+     * @param totalGrant             The total amount of tokens the user was granted
+     * @param merkleProof            The merkle proof showing the user is in the merkle tree
+     * @param lock                   The locking period to apply to the staked tokens
+     */
+    function claimAndStake(
+        address delegate,
+        uint128 totalGrant,
+        bytes32[] calldata merkleProof,
+        IAirdropSingleSidedStaking.Lock lock
+    ) external {
+        if (rewardsRoot == bytes32(0)) revert AA_NotInitialized();
+        // must be before the expiration time
+        if (block.timestamp > expiration) revert AA_ClaimingExpired();
+        // validate the withdraw
+        _validateWithdraw(totalGrant, merkleProof);
+
+        // approve the voting vault to transfer tokens
+        token.approve(address(votingVault), uint256(totalGrant));
+        // stake tokens in voting vault for this msg.sender and delegate
+        votingVault.airdropReceive(msg.sender, totalGrant, delegate, lock);
+
+        emit ClaimAndStaked(msg.sender, totalGrant, delegate, lock);
```

Comment:
Should probably just emit `Claim` here, all claims should emit a unified event or it adds overhead to all tracking/notification hooks (which may or may care if something is staked after or not).

You could emit two events, both `Claim` and `ClaimAndStake`, but IMO the latter is not needed because the staking contract will emit its own `Deposit` event